parallel.go

```diff
@@ -59,14 +59,22 @@ func (b *parallelBucketImpl) Push(ctx context.Context, local, remote string) err
 		go func() {
 			defer wg.Done()
 			for fn := range in {
-				catcher.Add(b.Bucket.Upload(ctx, filepath.Join(remote, fn), filepath.Join(local, fn)))
+				if b.dryRun {
+					continue
+				}
+
+				err = b.Bucket.Upload(ctx, filepath.Join(remote, fn), filepath.Join(local, fn))
+				if err != nil {
+					grip.Error(err)
+					catcher.Add(err)
```

Comment:
should we return? early? I wouldn't log in this case? or do so at a higher level?